x-pack/legacy/plugins/apm/server/routes/typings.ts

```diff
@@ -37,22 +43,29 @@ export interface Route<
   options?: {
     tags: Array<'access:apm' | 'access:apm_write'>;
   };
-  handler: (
-    req: Request,
-    params: DecodeParams<TParams>,
-    h: ResponseToolkit
-  ) => Promise<TReturn>;
+  handler: (kibanaContext: {
+    context: APMRequestHandlerContext<DecodeParams<TParams>>;
+    request: KibanaRequest;
+  }) => Promise<TReturn>;
 }
 
+export type APMRequestHandlerContext<
+  TDecodedParams extends { [key in keyof Params]: any } = {}
+> = RequestHandlerContext & {
+  params: { query: { _debug: boolean } } & TDecodedParams;
+  config: APMConfig;
+  logger: Logger;
+  __LEGACY: {
+    server: Server;
```

Comment:
nit: You're using a really broad type here which makes it hard to see which dependencies you're using. I would suggest just typing the dependencies you're using like:
```ts
__LEGACY: {
  server: {
    savedObjects: {
      getScopedSavedObjectsClient: Server['savedObjects']['getScopedSavedObjectsClient']
    }
    // similar for usage collector
  }
}
```

Reply:
Good one, I've narrowed the type to the things that we are using at the moment.